public/app/features/explore/Wrapper.tsx

```diff
@@ -118,6 +124,25 @@ class WrapperUnconnected extends PureComponent<Props, WrapperState> {
     document.title = documentTitle;
   }
 
+  windowResizeListener = debounce(() => {
+    let rightPaneRatio = 0.5;
+    const windowWidth = window.innerWidth;
+    // get the ratio of the previous rightPane to the window width
+    if (this.state.rightPaneWidth && this.state.windowWidth) {
+      rightPaneRatio = this.state.rightPaneWidth / this.state.windowWidth;
+    }
+    let newRightPaneWidth = Math.floor(windowWidth * rightPaneRatio);
+    if (newRightPaneWidth < this.minWidth) {
+      // if right pane is too narrow, make min width
+      newRightPaneWidth = this.minWidth;
+    } else if (windowWidth - newRightPaneWidth < this.minWidth) {
+      // if left pane is too narrow, make right pane = window - minWidth
+      newRightPaneWidth = windowWidth - this.minWidth;
+    }
+
+    this.setState({ windowWidth, rightPaneWidth: newRightPaneWidth });
```

Comment:
Works fine but I wonder if it'd be simpler to keep the ratio in the state instead of width, what do you think? This way we would not need to recalculate it here to adjust widths. What is more at the moment resizing feels a bit odd with only one pane being squeezed and not keeping the ratio.